packages/cosmic-swingset/test/run-policy.test.js

```diff
@@ -0,0 +1,200 @@
+/* eslint-env node */
+import test from 'ava';
+import { assert, q, Fail } from '@endo/errors';
+import { E } from '@endo/far';
+import { BridgeId, objectMap } from '@agoric/internal';
+import { makeFakeStorageKit } from '@agoric/internal/src/storage-test-utils.js';
+import {
+  defaultBootstrapMessage,
+  makeCosmicSwingsetTestKit,
+} from '../tools/test-kit.js';
+import { provideEnhancedKVStore } from '../src/helpers/bufferedStorage.js';
+import {
+  DEFAULT_SIM_SWINGSET_PARAMS,
+  makeVatCleanupBudgetFromKeywords,
+} from '../src/sim-params.js';
+
+/** @import { KVStore } from '../src/helpers/bufferedStorage.js' */
+
+/**
+ * Converts a Record<name, specifierString> into Record<name, { sourceSpec }>
+ * for defining e.g. a `bundles` group.
+ *
+ * @param {Record<string, string>} src
+ * @returns {import('@agoric/swingset-vat').SwingSetConfigDescriptor}
+ */
+const makeSourceDescriptors = src => {
+  const hardened = objectMap(src, sourceSpec => ({ sourceSpec }));
+  return JSON.parse(JSON.stringify(hardened));
+};
+
+/**
+ * @param {import('../src/sim-params.js').VatCleanupKeywordsRecord} budget
+ * @returns {import('@agoric/cosmic-proto/swingset/swingset.js').ParamsSDKType}
+ */
+const makeCleanupBudgetParams = budget => {
+  return {
+    ...DEFAULT_SIM_SWINGSET_PARAMS,
+    vat_cleanup_budget: makeVatCleanupBudgetFromKeywords(budget),
+  };
+};
+
+test('cleanup work must be limited by vat_cleanup_budget', async t => {
+  const { toStorage: handleVstorage } = makeFakeStorageKit('');
+  const receiveBridgeSend = (destPort, msg) => {
+    switch (destPort) {
+      case BridgeId.STORAGE: {
+        return handleVstorage(msg);
+      }
+      default:
+        Fail`port ${q(destPort)} not implemented for message ${msg}`;
+    }
+  };
+  const { pushCoreEval, runNextBlock, shutdown, swingStore } =
+    await makeCosmicSwingsetTestKit(receiveBridgeSend, {
+      bundles: makeSourceDescriptors({
+        puppet: '@agoric/swingset-vat/tools/vat-puppet.js',
+      }),
+      configOverrides: {
```

Comment:
Should this set a `defaultReapInterval`?

Reply:
Maybe when it's extended, but for now inheriting from `baseConfig` seems fine.